src/index.js

```diff
@@ -155,19 +162,49 @@ class Uploader {
         }
     }
 
-    prepareFilesForUpload(fileList) {
-        const fileIds = [],
-            fileCount = Object.keys(this.uploads).length,
-            maxFilesReached = Uploader.maxFilesReached(fileCount, this.maxFiles);
+    getOpenSlots() {
+        const fileCount = Object.keys(this.uploads).length;
+        const maxFilesReached = Uploader.maxFilesReached(fileCount, this.maxFiles);
+
+        if (maxFilesReached) { return 0 };
+
+        return this.maxFiles - fileCount;
+    }
+
+    maxSizeExceeded(size) {
+        if (!this.maxFileSize) { return false; }
+        return this.maxFileSize < Uploader.bytesToMegabytes(size)
+    }
+
+    fileAddRestrictions({ openSlots, fileMeta = {}}) {
```

Comment:
this function does 2 things, split it to fileValidations and "uploader" like validations